src/shared/specifier.c

```diff
@@ -175,6 +175,19 @@ int specifier_uuid(char specifier, const void *data, const char *root, const voi
         return 0;
 }
 
+int specifier_uint64(char specifier, const void *data, const char *root, const void *usedata, char **ret) {
+        const uint64_t *n = ASSERT_PTR(data);
+        char *s;
+        int r;
+
+        r = asprintf(&s, "%" PRIu64, *n);
+        if (r < 0)
+                return -errno;
+
+        *ret = s;
```

Comment:
also, you can get rid of `s` too, and let asprintf() write directly to `ret`. This is a glibc specific API after all, and in glibc it doesn't touch the return parameter on failure (like a good boy), so we can just allow it to write to ret directly.